### DISCUSSION TITLE
Model json field fails in Livewire test

### DISCUSSION THREAD

**Discussion by matteo-palazzo** [livewire/livewire#5415] [1. Help]
Hi guys!

Inside my Livewire component I print a json property of a Model casted to array. In the component everything works as expected, the json is casted to array and is printed just fine. The problem is the test, it seems that the json is not casted to array and a 500 is returned: "Cannot access offset of type string on string".

```html
@livewire('my-component', ['user' => $user])

{{ $user->options['color'] }}
```

```php
class MyComponent extends Component
{
    public User $user;

    public function mount(User $user)
    {
        $this->user = $user;
    }
}
```

```php
class User extends Model
{
    protected $attributes = [
        'options' => '{"color":"red"}'
    ];

      protected $casts = [
           'options' => 'array',
    ];
}
```

```php
/** @test */
public function options_not_working_in_test()
{
    $this->withoutExceptionHandling();

    $user = User::factory()->create();

    Livewire::test('options', ['user' => $user])
        ->assertSeeText('red');
}
```

Am I missing something? I'm not that accustomed to Livewire, maybe I'm misusing it.

**Comment by joshhanley** [comment]
@matteo-palazzo can you share a screenshot of the error?

**Comment by matteo-palazzo** [comment]
Yeah, of course! Here it is. I'm on latest Laravel and Livewire with PHP8.1. If you'd like to reproduce the error the code above is everything you need. Thanks!

![Schermata del 2022-12-12 14-46-30](https://user-images.githubusercontent.com/83695398/207061506-99c2f185-e2c5-4e20-be90-9ce5b19e37bd.png)

**Comment by joshhanley** [comment]
@matteo-palazzo sorry about the delay, only just got a chance to look at this.

I couldn't replicate your issue for me it all worked happily and the test passed.

I did notice though that you have manually set `$attributes` on the user model, maybe that was causing the issue, so try changing that to load data from the database and see if it works.

Hope this helps!
